integration/watch.ts

```diff
@@ -1,86 +1,150 @@
+  const yarn = process.platform === 'win32' ? 'yarn.cmd' : 'yarn';
+  const usePolling = process.argv.includes('--polling');
+  const tests = process.argv.slice(2).filter((x) => !x.startsWith('-'));
+  const watchOptions = { ignoreInitial: true, usePolling };
+
+  if (process.argv.includes('-h') || process.argv.includes('--help')) showHelp();
+  showSettings(tests);
+
+  chokidar
+    .watch('integration/*/*.proto', watchOptions)
+    .on('all', integrationHandler(yarn, 'proto2bin'));
+
+  chokidar
+    .watch('integration/*/*.proto', watchOptions)
+    .on('all', integrationHandler(yarn, 'proto2pbjs'));
+
+  chokidar
+    .watch('integration/*/*.bin', watchOptions)
+    .on('all', integrationHandler(yarn, 'bin2ts'));
+
+  chokidar
+    .watch('src/**/*.ts', watchOptions) //
+    .on('all', sourceHandler(yarn, 'bin2ts', tests));
+
+  setupKeys({
+    ['']: () => yarnRun(yarn, 'bin2ts', 'enter'),
+    ['q']: () => process.exit(),
+    ['\u0003']: () => process.exit(), // ctrl-c
+  });
+}
+
+function setupKeys(bindings: { [p: string]: () => void }) {
+  process.stdin.setRawMode(true);
+  process.stdin.resume();
+  process.stdin.setEncoding('utf8');
 
-    if (process.argv.includes('-h') || process.argv.includes('--help')) {
-        console.log(`
+  Object.entries(bindings).forEach(([key, handler]) => {
+    process.stdin.on('data', (data) => {
+      if (data.toString().trim() === key) handler();
+    });
+  });
+}
+
+function showHelp() {
+  console.log(`
 Watches the integration directory for changes and regenerates .bin and .ts files.
+Watches the src directory and regenerates given TEST files.
 
 Usage:
-    $ yarn watch 
-    $ ts-node watch.ts [options]
+    $ yarn watch [options] [TEST, TEST2, ...]
+    $ ts-node watch.ts [options] [TEST, TEST2, ...]
     
 Options:
-    -h, --help    Show this help message
-    --polling     Use polling instead of native watchers
-        `)
-        process.exit(0)
-    }
-    const usePolling = process.argv.includes('--polling')
-
-    process.chdir("integration");
+    -h, --help    Show this help message.
+    --polling     Use polling instead of native watchers.
+    TEST          Recompile the specified TEST(s) when implementation files change.
+                  Equivalent to running 'yarn bin2ts TEST' manually each time.
+    
+Examples:
+    $ yarn watch
+    $ yarn watch --polling
+    $ yarn watch simple struct
+        `);
+  process.exit(0);
+}
 
-    chokidar
-      .watch("*/*.proto", { ignoreInitial: true, usePolling  })
-      .on('all', yarnRunHandler(yarn, 'proto2bin'));
+function showSettings(tests: string[]) {
+  console.log(`${colors.yellow}`);
+  console.log(`Watching for changes:`);
+  console.log('- integration/\tRecompiling .proto to .bin and .ts');
+  if (tests.length) {
+    console.log(`- src/\t\tRecompiling .proto to .ts\n\t\t(${tests.join(', ')})`);
+  }
+  console.log('\nPress enter to recompile all .proto files.');
```

Comment:
Maybe "to regenerate all integration tests"? Or "to codegen all integration tests", granted I like the term codegen and regenerate is probably better.